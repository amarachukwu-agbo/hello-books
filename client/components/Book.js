import React from 'react';
import { Link } from 'react-router-dom';

const Book = (props) => {
  const { book } = props;
  return (
        <div className="col s12 m6 l4">
            <div className="card">
                <div className="card-image">
                    <img src = { book.imageURL } className="responsive-img" />
                    <Link to= { `/books/${book.id}`} className="btn-floating halfway-fab waves-effect waves-light small red">
                        <i className="material-icons">chevron_right</i>
                    </Link>
                </div>
                <div className="card-content">
                    <span className="card-title truncate grey-text text-darken-4"> { book.title } </span>
                    <p> Author| <span className="red-text text-darken-4 bold"> { book.author} </span></p>
                    <p> Genre| <span className="red-text text-darken-4 bold"> {book.subject} </span></p>
                    <p> Quantity| <span className="red-text text-darken-4 bold"> { book.quantity} </span></p>
                    <div className="divider"></div>
                    <div className="row">
                        <div className="col s3 btn btn-small btn-flat white teal-text book-icon"><i className="material-icons prefix">thumb_up</i><span>{book.upvotes}</span></div>
                        <div className="col s3 btn btn-small btn-flat white teal-text book-icon"><i className="material-icons prefix">favorite_border</i><span>{book.favCount}</span></div>
                        <div className="col s3 btn btn-samll btn-flat white teal-text book-icon"><i className="material-icons prefix">comment</i><span>{book.bookReviews.length}</span></div>
                    </div>
                </div>
            </div>
        </div>
  );
};

export default Book;
