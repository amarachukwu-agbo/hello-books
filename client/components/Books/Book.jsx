import React from 'react';
import propTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * @description stateless component for rendering a book
 *
 * @param {object} props - properties of the component
 *
 * @returns {Node} - react node containing the book component
 */
const Book = (props) => {
  const { book } = props;
  return (
    <div className="col s6 m4 l3">
      <div className="card">
        <div className="card-image">
          <img src={book.imageURL} className="responsive-img" />
          <Link to={`/books/${book.id}`}
            className="btn-floating halfway-fab waves-effect
            waves-light small red">
            <i className="material-icons hoverable">chevron_right</i>
          </Link>
        </div>
        <div className="card-content">
          <span className="card-title truncate grey-text text-darken-4">
            {book.title}
          </span>
          <p className="truncate">Author|
            <span className="red-text text-darken-4 bold">{book.author}</span>
          </p>
          <p className="truncate"> Genre|
            <span className="red-text text-darken-4 bold">{book.subject}</span>
          </p>
          <p> Quantity|
            <span className="red-text text-darken-4 bold">{book.quantity}</span>
          </p>
          <div className="divider hide-on-med-and-down"></div>
          <div className="row hide-on-med-and-down">
            <div className="col s3 btn btn-small btn-flat white grey-text
            text-darken-2 book-icon">
              <i className="material-icons prefix">thumb_up</i>
              <span>{book.upvotes}</span></div>
              <div className="col s3 btn btn-small btn-flat white
              grey-text text-darken-2 book-icon">
                <i className="material-icons prefix">favorite_border</i>
                <span>{book.favCount}</span>
              </div>
            <div className="col s3 btn btn-samll btn-flat white
              grey-text text-darken-2 book-icon">
              <i className="material-icons prefix">comment</i>
              <span>{book.bookReviews.length}</span></div>
          </div>
        </div>
      </div>
    </div>
  );
};

// Prop type validation
Book.propTypes = {
  book: propTypes.shape({
    author: propTypes.string.isRequired,
    subject: propTypes.string.isRequired,
    bookReviews: propTypes.array.isRequired,
    favCount: propTypes.number.isRequired,
    upvotes: propTypes.number.isRequired,
    downvotes: propTypes.number.isRequired,
    quantity: propTypes.number.isRequired,
    imageURL: propTypes.string.isRequired,
  }),
};

export default Book;
