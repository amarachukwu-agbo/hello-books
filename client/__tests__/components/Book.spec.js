import React from 'react';
import renderer from 'react-test-renderer';
import Book from '../../components/Books/Book';

const props = {
  book: {
    title: 'Magnolia Table: A Collection of Recipes for Gathering',
    author: 'Joanna Gaines',
    description: 'Book for test',
    subject: 'Recipe',
    imageURL: 'https://images-na.3,200_.jpg',
    quantity: 6,
    favCount: 2,
    upvotes: 1,
    downvotes: 2,
    bookReviews: [
      {
        id: 3,
        review: 'I love it',
        createdAt: '2018-06-27T12:18:05.917Z',
        updatedAt: '2018-06-27T12:18:05.917Z',
        userId: 12,
        bookId: 98,
        userReviews: {
          id: 12,
          firstName: 'Amarachi',
          lastName: 'Agbo',
          email: 'amarachukwu.agbo@gmail.com',
          role: 'User',
          imageURL: null,
          createdAt: '2018-06-27T12:01:52.036Z',
          updatedAt: '2018-06-27T12:01:52.036Z',
        },
      },
    ],
  },
};
describe('<Book>', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Book { ...props }/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
