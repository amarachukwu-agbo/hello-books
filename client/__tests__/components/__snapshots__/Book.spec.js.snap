// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book Component> should render correctly 1`] = `
<div
  className="col s6 m4 l3"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        className="responsive-img"
        src="https://images-na.3,200_.jpg"
      />
      <a
        className="btn-floating halfway-fab waves-effect waves-light small red"
        href="/books/undefined"
      >
        <i
          className="material-icons hoverable"
        >
          chevron_right
        </i>
      </a>
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title truncate grey-text text-darken-4"
      >
        Magnolia Table: A Collection of Recipes for Gathering
      </span>
      <p
        className="truncate"
      >
        Author|
        <span
          className="red-text text-darken-4 bold"
        >
          Joanna Gaines
        </span>
      </p>
      <p
        className="truncate"
      >
         Genre|
        <span
          className="red-text text-darken-4 bold"
        >
          Recipe
        </span>
      </p>
      <p>
         Quantity|
        <span
          className="red-text text-darken-4 bold"
        >
          6
        </span>
      </p>
      <div
        className="divider hide-on-med-and-down"
      />
      <div
        className="row hide-on-med-and-down"
      >
        <div
          className="col s3 btn btn-small btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            thumb_up
          </i>
          <span>
            1
          </span>
        </div>
        <div
          className="col s3 btn btn-small btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            favorite_border
          </i>
          <span>
            2
          </span>
        </div>
        <div
          className="col s3 btn btn-samll btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            comment
          </i>
          <span>
            1
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Book> should render correctly 1`] = `
<div
  className="col s6 m4 l3"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        className="responsive-img"
        src="https://images-na.3,200_.jpg"
      />
      <a
        className="btn-floating halfway-fab waves-effect waves-light small red"
        href="/books/undefined"
      >
        <i
          className="material-icons hoverable"
        >
          chevron_right
        </i>
      </a>
    </div>
    <div
      className="card-content"
    >
      <span
        className="card-title truncate grey-text text-darken-4"
      >
        Magnolia Table: A Collection of Recipes for Gathering
      </span>
      <p
        className="truncate"
      >
        Author|
        <span
          className="red-text text-darken-4 bold"
        >
          Joanna Gaines
        </span>
      </p>
      <p
        className="truncate"
      >
         Genre|
        <span
          className="red-text text-darken-4 bold"
        >
          Recipe
        </span>
      </p>
      <p>
         Quantity|
        <span
          className="red-text text-darken-4 bold"
        >
          6
        </span>
      </p>
      <div
        className="divider hide-on-med-and-down"
      />
      <div
        className="row hide-on-med-and-down"
      >
        <div
          className="col s3 btn btn-small btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            thumb_up
          </i>
          <span>
            1
          </span>
        </div>
        <div
          className="col s3 btn btn-small btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            favorite_border
          </i>
          <span>
            2
          </span>
        </div>
        <div
          className="col s3 btn btn-samll btn-flat white grey-text text-darken-2 book-icon"
        >
          <i
            className="material-icons prefix"
          >
            comment
          </i>
          <span>
            1
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
